playwright/tests/e2e/steps/transactions-overview.steps.ts

```diff
@@ -0,0 +1,221 @@
+    const trxHash = await page.getByTestId("transaction.table.value.txhash#0");
+    const ariaLabelValue = await trxHash?.getAttribute("aria-label");
+    if (ariaLabelValue) {
+      (await koiosApi(request).getTransactionByTrxHash(ariaLabelValue)).json().then(async (data) => {
+        await transactionOverviewPage(page).checkCurrenTransactionWidget({ currentTransaction: data });
+      });
+    }
+  }
+);
+
+Given(/^the user is in the Transactions page$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects the transaction hash of one record of the transactions table in the transactions page$/,
+  async ({ page }) => {
+    const transactionsTableValueEpoch = page.getByTestId("transaction.table.value.txhash#0");
+    transactionHash = await transactionsTableValueEpoch.getAttribute("href");
+    await transactionsTableValueEpoch.click();
+  }
+);
+Then(/^the transaction details page of the select transaction should be opened$/, async ({ page }) => {
+  await transactionDetailPage(page).checkTransactionsDetail({ transactionHash: transactionHash || "" });
+});
+
+Given(/^the user is in the Transactions page portal$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects the block number of one record of the transactions table in the transactions page$/,
+  async ({ page }) => {
+    const transactionsTableValueEpoch = page.getByTestId("transaction.table.value.block#0");
+    await transactionsTableValueEpoch.click();
+  }
+);
+Then(/^the user should be redirected to the block detail page of the selected block number$/, async ({ page }) => {
+  const deatailPageTitle = page.getByTestId("detail.page.title");
+  await expect(deatailPageTitle, "Check title on block detail").toHaveText("Block Details");
+});
+
+Given(/^the user is in the Transactions page web$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects the epoch number of one record of the transactions table in the transactions page$/,
+  async ({ page }) => {
+    const transactionsTableValueEpoch = page.getByTestId("transactions.table.value.epoch#0");
+    await transactionsTableValueEpoch.click();
+  }
+);
+Then(/^the user should be redirected to the epoch detail page of the selected epoch number$/, async ({ page }) => {
+  const deatailPageTitle = page.getByTestId("detail.page.title");
+  await expect(deatailPageTitle, "Check title on epoch detail").toHaveText("Epoch Details");
+});
+
+Given(/^the user is in the Transactions page for go to address detail page$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects the input or output address of one record of the transactions table in the transactions page$/,
+  async ({ page }) => {
+    const transactionsTableValueEpoch = page.getByTestId("transaction.table.value.inputAddress#0").first();
```

Comment:
let's keep locators interactions inside the pages functions and returns the values that you could need in the next steps if need it.